D3D12Bundles/D3D12Bundles.cs

```diff
@@ -699,43 +651,41 @@ struct Vertex {
             // However, when ExecuteCommandList() is called on a particular command
             // list, that command list can then be reset at any time and must be before
             // re-recording.
-            mCommandList.Reset(mCurrentFrameResource.mCommandAllocator, mPipelineState1);
+            commandList.Reset(mCurrentFrameResource.mCommandAllocator, mPipelineState1);
```

Comment:
Here we use one of the new methods, not part of DX12GE as a workaround to retain this behaviour of the D3D12Bundles application.

CommandLists in DX12GE use their own allocators and always set the root signature immediately before the pipeline state, which is incompatible with out requirements to not group that behaviour together here.

Maybe in time I can refactor to remove the need for this once I grow in understanding about the importance of certain orders of events or the impact of doing certain operations twice.

e.g. if I set the root signature again each time I need to set the pipeline state, or vice-versa, set hte pipeline state each time I need to set the root signature, does that have a performance impact?  Or does it matter if I set the descriptor heaps multiple times to different values?  If none of this matters I can use the DX12GE Reset method safely, but until I understand that, I've exposed this to do only what we know is safe to do, as intended by the MS developers that wrote the sample.